pkg/pillar/types/zedmanagertypes.go

```diff
@@ -132,6 +133,36 @@ type AppInstanceStatus struct {
 	ErrorAndTimeWithSource
 }
 
+func (status AppInstanceStatus) LogCreate() {
+	logObject := base.NewLogObject(base.AppInstanceLogType, status.DisplayName,
+		status.UUIDandVersion.UUID, status.Key())
+	if logObject == nil {
+		return
+	}
+	logObject.CloneAndAddField("state", status.State).AddField("restart-in-progress", status.RestartInprogress).
+		AddField("purge-in-progress", status.PurgeInprogress).Infof("App instance status")
+}
+
+func (status AppInstanceStatus) LogModify(old interface{}) {
+	logObject := base.LookupLogObject(status.Key())
+	if logObject == nil {
+		logObject = base.NewLogObject(base.AppInstanceLogType, status.DisplayName,
+			status.UUIDandVersion.UUID, status.Key())
```

Comment:
I think will see a fair bit of these and it might make sense to have a lookupOrCreate() or a ensureLogObject() (the latter would panic if not found and it can't allocate it).

Reply:
will do EnsureLogObject and InitLogObject will panic if log object allocation fails.